app/controllers/posts_controller.rb

```diff
@@ -18,7 +18,7 @@ def create
   end
 
   def show
-    @post = Post.find(params[:id])
+    @post = Post.includes(:author).find(params[:id])
   end
```

Comment:
- [ ] Kindly fix the N + 1 query problem in the index action, not in the show action. Please make sure that you are fetching posts together with their comments, so in this case, a post should come with its comments and not its `author`.